xbmc/platform/linux/input/LibInputHandler.cpp

```diff
@@ -107,9 +108,30 @@ void CLibInputHandler::Start()
 
 void CLibInputHandler::Process()
 {
+  int epollFd = epoll_create1(0);
+  if (epollFd < 0)
+  {
+    CLog::Log(LOGERROR, "CLibInputHandler::%s - failed to create epoll file descriptor: %s", __FUNCTION__, strerror(-errno));
+    return;
+  }
+
+  epoll_event event;
+  event.events = EPOLLIN;
+  event.data.fd = m_liFd;
+
+  auto ret = epoll_ctl(epollFd, EPOLL_CTL_ADD, m_liFd, &event);
+  if (ret < 0)
+  {
+    CLog::Log(LOGERROR, "CLibInputHandler::%s - failed to add file descriptor to epoll: %s", __FUNCTION__, strerror(-errno));
+    close(epollFd);
+    return;
+  }
+
   while (!m_bStop)
   {
-    auto ret = libinput_dispatch(m_li);
+    epoll_wait(epollFd, &event, 1, -1);
+
+    ret = libinput_dispatch(m_li);
     if (ret < 0)
     {
       CLog::Log(LOGERROR, "CLibInputHandler::%s - libinput_dispatch failed: %s", __FUNCTION__, strerror(-errno));
```

Comment:
you're leaking the `epollFd` here on error, this should be the usual `goto out:` or whatever other approach you prefer :)

Reply:
Thanks for catching that!